Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,8 +193,74 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
-            return array;
+
+            if (!IsValidCaseForInsertForNullOrEmpty(array, index))
+            {
+                return array;
+            }
+
+            if (! IsValidCaseForInsertForNullOrEmpty(array, number, index)  )
+            {
+                if (index == 0)
+                {
+                    return new [] { number };
+                }
+                return array; 
+            }
+
+           
+            
+            var newArray = Insert(array, number, index);
+
+            return newArray;
+
+        }
+
+        public static bool IsValidCaseForInsertForNullOrEmpty(int[] array, int index)
```

Comment:
You don't use array here. You don't need it to be a parameter then. However you should probably delete this anyway.

Reply:
?? if I delete that as parameter. I see a lot of red swiggeling lines about that array is not found.

Reply:
I guess that is because you put in array in line 197. The only variable you use within this function is index so you only need that one as parameter.